File: 백준/Gold/1438. 가장 작은 직사각형/가장 작은 직사각형.py
"""
완탐
어떤 좌표가 주어지면 거기서 가능성이 있는 y 좌표들
그럼 후보는 y좌표 100개를 중복 조합으로 뽑기 5050

그리고 그 안의 좌표가 N//2개를 못넘으면 다음으로 넘어가고

넘는다면 모든 경우마다 사각형의 넓이 구해보기
"""
import sys

input = lambda: sys.stdin.readline().strip()

n = int(input())

lst = []
for _ in range(n):
    lst.append(list(map(int, input().split())))

lst = sorted(lst)
answer = 100_000_000
for i in range(n):
    for j in range(i, n):
        can = []
        for y, x in lst:
            if lst[i][0] <= y <= lst[j][0]:
                can.append(x)
        if len(can) < n / 2:
            continue
        can = sorted(can)
        for k in range(int(len(can) - n / 2) + 1):
            answer = min(answer, (abs(lst[i][0] - lst[j][0]) + 2) * (can[k + n // 2 - 1] - can[k] + 2))

print(answer)

File: 백준/Gold/1438. 가장 작은 직사각형/가장 작은 직사각형.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static StringTokenizer st;
    static StringBuilder sb = new StringBuilder();

    public static void main(String[] args) throws IOException {

        int N = Integer.parseInt(br.readLine());
        int[][] lst = new int[N][2];

        for (int i = 0; i < N; i++) {
            st = new StringTokenizer(br.readLine());
            lst[i][0] = Integer.parseInt(st.nextToken());
            lst[i][1] = Integer.parseInt(st.nextToken());
        } // 입력완료

        Arrays.sort(lst, Comparator.comparingInt((int[] o) -> o[0]).thenComparingInt(o -> o[1]));
        int answer = 100_000_000;
        for (int i = 0; i < N; i++) {
            for (int j = i; j < N; j++) {
                List<Integer> can = new ArrayList<>();
                for (int k = 0; k < N; k++) {
                    if (lst[i][0] <= lst[k][0] && lst[k][0] <= lst[j][0]) {
                        can.add(lst[k][1]);
                    }
                }
                if (can.size() < N / 2)
                    continue;
                Collections.sort(can);
                for (int k = 0; k < can.size() - (int) (N / 2) + 1; k++) {
                    answer = Math.min(answer, (lst[j][0] - lst[i][0] + 2) * (can.get(k + (int) (N / 2) - 1) - can.get(k) + 2));
                }
            }
        }

        System.out.println(answer);

    } // end of main

} // end of class